### PR TITLE
Fix: duplicated ClientManyProductsQuery

### PR DESCRIPTION
**Pull request by lariciamota**
## What's the purpose of this pull request?

The `ClientManyProductsQuery` was being fetched twice with almost the same parameters, the only difference was the `hasOnlyDefaultSalesChannel` introduced on this [PR](https://github.com/vtex/faststore/pull/2282) and used only by the `useAuth` hook. 
It doesn't need to be sent on the `ClientManyProductsQuery`, only on the `ValidateSession` mutation.
These other queries were also being affected: `ClientSearchSuggestionsQuery`, `ClientProductGalleryQuery` and `ClientProductQuery`.

## How it works?

It filters the `hasOnlyDefaultSalesChannel` key out of the `channel` object in the `useSession` before sending it on the query.

## How to test it?

Running `yarn dev` and checking on the PDP that the `ClientManyProductsQuery` is not duplicated and is not sending the `hasOnlyDefaultSalesChannel` on `channel`.
| Before | After |
| -|-|
| Screenshot 2024-09-02 at 14 00 24 | Screenshot 2024-09-02 at 13 58 30 |
| Screenshot 2024-09-02 at 14 00 54 | |

## References

- [Slack thread](https://vtex.slack.com/archives/C051B6LL91U/p1725284216284089)

### CODE REVIEW

**Review by pedromtec — APPROVED** [review]

**Review by eduardoformiga — APPROVED** [review]

**Review by mateussaggin — CHANGES_REQUESTED** [review]
I think it's not 100% fixed. Now the fuzzy is the only difference in the query:
Example below

image

https://github.com/user-attachments/assets/362af7cf-c888-4786-86b8-212d94ee368b

**Comment by lariciamota** [comment]
> I think it's not 100% fixed. Now the fuzzy is the only difference in the query: Example below
> 
> image

Hi! The fuzzy is being tackled in another PR that will handle the issues introduced by the fuzzy (this included)

File: packages/core/src/sdk/session/index.ts [hunk 1]
```diff
@@ -69,15 +69,31 @@ export const sessionStore = {
...
-  const session = useStore(sessionStore)
+/**
+ * This custom hook is used to retrieve the session data and filter the channel object.
+ * The channel object filtering removes the hasOnlyDefaultSalesChannel key.
+ * This key is used only in the useAuth hook and is only required to send on the ValidateSession mutation,
+ * so we remove it from the session's channel object to avoid unnecessary cache invalidations and query executions.
+ *
+ * @param filter - A boolean value indicating whether to filter the channel object or not. Default is true.
+ * @returns An object containing the session data, channel object, and a flag indicating whether the session is being validated.
+ */
```
Comment (ArthurTriis1) [inline packages/core/src/sdk/session/index.ts:80]: Thank you for description!

**Review by ArthurTriis1 — APPROVED** [review]

File: packages/core/src/sdk/session/index.ts [hunk 2]
```diff
@@ -69,15 +69,31 @@ export const sessionStore = {
...
+export const useSession = (filter: boolean = true) => {
```
[the remark this answers is not in the document]
Comment (lariciamota) [reply packages/core/src/sdk/session/index.ts:81]: Good idea!

**Review by pedromtec — APPROVED** [review]

File: packages/core/src/sdk/session/index.ts [hunk 3]
```diff
@@ -69,15 +69,37 @@ export const sessionStore = {
...
+ * The channel object filtering removes the hasOnlyDefaultSalesChannel key.
+ * This key is used only in the useAuth hook and is only required to send on the ValidateSession mutation,
+ * so we remove it from the session's channel object to avoid unnecessary cache invalidations and query executions.
+ *
+ * @param options - Optional configuration for the hook.
+ * @param options.filter - A boolean value indicating whether to filter the channel object or not. Default is true.
+ * @returns An object containing the session data, channel object, and a flag indicating whether the session is being validated.
+ */
+
+export const useSession = ({ filter }: SessionOptions = { filter: true }) => {
```
Comment (eduardoformiga) [inline packages/core/src/sdk/session/index.ts:87]: Nice! I think in the future, if we have more fields to filter, we should pass an array of keys, but so far, it looks good to me!

File: packages/core/src/sdk/session/index.ts (hunk 1 above)
Comment (eduardoformiga) [reply packages/core/src/sdk/session/index.ts]: Nice! 💯

Comment (hellofanny) [reply packages/core/src/sdk/session/index.ts]: thanks for this 🌟

**Pull request merged**